src/substrait/textplan/converter/HierarchyStack.h

```diff
@@ -0,0 +1,44 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include <google/protobuf/message.h>
+
+namespace io::substrait::textplan {
+
+class HierarchyStack {
+ public:
+  void push(const ::google::protobuf::Message* protoAddr) {
```

Comment:
What's your logic / guideline for when to put something in a header vs a cc file?

Reply:
Triviality, I'm hitting my perceived line here but I agree it does look a little weird having one function by itself in a file.  I've moved everything into the header.